lib/utils/forkedblockchain.js

```diff
@@ -107,7 +101,7 @@ ForkedBlockchain.prototype.createGenesisBlock = function(callback) {
 
     // Update the relevant block numbers
     self.forkBlockNumber = self.options.fork_block_number = blockNumber;
-    self.stateTrie.forkBlockNumber = blockNumber;
+    self.stateTrie.forkBlockNumber = self.stateTrie.options.forkBlockNumber = blockNumber;
```

Comment:
Reason for change: The `stateTrie` is now `copy`ed instead of created via `new ForkedStorageTrie` (this aligns with how the merkle patricia trie does it internally). When copying we need to also copy the options object, so here I am making sure the forkBlockNumber is set on the object.